spec/models/swift_storage_provider_spec.rb

```diff
@@ -1,13 +1,462 @@
+          "x-account-meta-temp-url-key" => subject.primary_key,
+          "x-account-meta-temp-url-key-2" => subject.secondary_key
+        }}
+        it 'should raise StorageProviderException' do
+          is_expected.to receive(:get_account_info)
+            .and_return(account_info)
+          expect {
+            expect(subject.is_ready?).to be_truthy
+          }.not_to raise_error
+        end
+      end
+    end
+
+    describe '#initialize_project' do
+      it 'should create a container in swift with the project id' do
+        expect(project).to receive(:id)
+          .and_return(expected_project_id)
+        is_expected.to receive(:put_container)
+          .with(expected_project_id)
+        expect {
+          subject.initialize_project(project)
+        }.not_to raise_error
+      end
+    end
+
+    describe '#is_initialized?(project)' do
+      before do
+        expect(project).to receive(:id)
+          .and_return(expected_project_id)
+      end
+
+      context 'project container exists' do
+        let(:expected_meta) {
+          {
+            foo: 'bar',
+            baz: 'stuff'
+          }
+        }
+        it 'should return true' do
+          is_expected.to receive(:get_container_meta)
+            .with(expected_project_id)
+            .and_return(expected_meta)
+          expect {
+            expect(subject.is_initialized?(project)).to be_truthy
+          }.not_to raise_error
+        end
+      end
+
+      context 'project container does not exist' do
+        it 'should return false' do
+          is_expected.to receive(:get_container_meta)
+            .with(expected_project_id)
+          expect {
+            expect(subject.is_initialized?(project)).to be_falsey
+          }.not_to raise_error
+        end
+      end
+
+      context 'unexpected StorageProviderException' do
+        let(:unexpected_exception) { StorageProviderException.new('Unexpected') }
+        it 'should raise the original exception' do
+          is_expected.to receive(:get_container_meta)
+            .with(expected_project_id)
+            .and_raise(unexpected_exception)
+          expect {
+            subject.is_initialized?(project)
+          }.to raise_error(unexpected_exception)
+        end
+      end
+    end
+
+    describe '#single_file_upload_url(upload)' do
+      let(:expected_url) { Faker::Internet.url }
+      it 'should return a signed url to POST the upload' do
+        is_expected.to receive(:build_signed_url)
+          .with(
+            'POST',
+            upload.sub_path,
+            subject.expiry
+          ).and_return(expected_url)
+        expect {
+          expect(subject.single_file_upload_url(upload)).to eq(expected_url)
+        }.not_to raise_error
+      end
+    end
+
+    describe '#initialize_chunked_upload' do
+      it 'should not do anything to initialize a chunked upload in swift' do
+        expect {
+          subject.initialize_chunked_upload(upload)
```

Comment:
it does not do anything at all (it just returns) in the swift implementation since it isn't required. I can have it return true if you want. Then when we implement the S3 implementation it will need to return true or raise an exception.